### ISSUE TITLE
windows agents not showing online 2024

### ISSUE THREAD

**Issue by adnan29979** [Ylianst/MeshCentral#5765]
I've been running Windows 11 agents on two separate MeshCentral servers without any hiccups. But starting from January 31, 2024, they've just stopped showing up online. At first, I figured my team hadn't turned their devices on, but after checking, it turns out the devices are indeed powered up and running. The MeshAgent service is active in the background on both, yet for some reason, they're no longer appearing online on either MeshCentral server since that date.
At the same time, my Windows 10 agents are running just fine. 
![Screenshot 2024-02-02 203106](https://github.com/Ylianst/MeshCentral/assets/148310766/96eadb05-4548-44fd-879f-ebebbc01d123)
![Screenshot 2024-02-02 203123](https://github.com/Ylianst/MeshCentral/assets/148310766/46ef87fa-fa55-4e44-903b-cf1921a89f34)

**Comment by si458** [comment]
you might be effected by the bitlocker bug that happened in the previous release!
one think you can try is replace `node_modules/meshcentral/agents/modules_meshcore/computer-identifiers.js` with the master JS file,https://github.com/Ylianst/MeshCentral/blob/master/agents/modules_meshcore/computer-identifiers.js
once done, then restart your meshcentral server

**Comment by adnan29979** [comment]
replaced the file computer-identifiers.js. The issue persists. The Windows 11 did not come back online (the Windows 11 machine is up and running)

**Comment by adnan29979** [comment]
Apologies. 

Config.json:
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "my.domain",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 2048,
    "_aliasPort": 443,
    "redirPort": 80,
    "redirAliasPort": 80,
    "MongoDB": "mongodb://127.0.0.1:27017/?replicaSet=rs0",
    "MongoDbBulkOperations": true,
    "selfUpdate": false,
    "webRTC": true,
    "_agentsInRam": true,
    "dbExpire": {
      "events": 10368000,
      "powerevents": 10368000,
      "statsevents": 10368000
    },
    "maxInvalidLogin": {
      "time": 12,
      "count": 2,
      "coolofftime": 5
    },
    "maxInvalid2fa": {
      "time": 12,
      "count": 2,
      "coolofftime": 5
    },
    "_crowdsec": { "fallbackRemediation": "captcha" },
    "AutoBackup": {
      "mongoDumpPath": "/usr/bin/mongodump",
      "backupIntervalHours": 24,
      "keepLastDaysBackup": 5,
      "zipPassword": "mypass",
      "_backupPath": "C:\\backups",
      "_googleDrive": {
        "folderName": "MeshCentral-Backups-GDrive",
        "maxFiles": 5
      },
    "_ipBlockedUserRedirect": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
    "_userAllowedIP": "127.0.0.1,192.168.1.0/24",
    "_userBlockedIP": "127.0.0.1,::1,192.168.0.100",
    "agentPing": 18,
    "_browserPing": 38
    }
  },
  "domains": {
    "": {
      "_title": "mytitle",
      "title": "mytitle",
      "_title2": "Servername",
      "_minify": true,
      "newAccounts": false,
      "_userNameIsEmail": true,
      "twoFactorCookieDurationDays": 12,
      "_titlePicture": "logo.png",
      "welcomePicture": "logo.png",
      "agentCustomization": {
        "displayName": "xxxxx",
        "description": "xxxxx",
        "companyName": "xxxxx",
        "serviceName": "xxxxx",
        "image": "xxxxx.png",
        "fileName": "xxxxx",
        "foregroundColor": "#FFFFFF",
        "_backgroundColor": "#xxxxx",
	"backgroundColor": "#xxxxx",
        "installText": "xxxxx"
        },
      "agentFileInfo": {
       "icon": "xxxxx.ico",
       "filedescription": "xxxxx",
       "fileversion": "12.0.4.1",
       "internalname": "xxxxx",
       "legalcopyright": "xxxxx",
       "originalfilename": "xxxxx",
       "productname": "xxxxx",
       "productversion": "v12.0.4.1"
      },
      "androidCustomization": {
      "title": "xxxxx",
      "subtitle": "xxxxx",
      "_image": ".png"
      },
      "_terminal": {
        "_linuxshell": "login",
        "launchCommand": {
          "linux": "clear\nsu pi\ncd\n",
          "_darwin": "clear\necho \"Hello MacOS\"\n",
          "_freebsd": "clear\necho \"Hello FreeBSD\"\n"
        }
      }
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "xxxxx@xxxxx.com",
    "names": "xxxxx",
    "skipChallengeVerification": true,
    "production": true
  },
  "smtp": {
    "host": "smtp.gmail.com",
    "port": 465,
    "from": "xxxxx@xxxxx.com",
    "auth": {
      "clientId": "xxxxx",
      "clientSecret": "xxxxx",
      "refreshtoken": "xxxxx"
    },
    "user": "xxxxx@xxxxx.com",
    "tls": false,
    "verifyEmail": true
  }
}

Server OS:
Debian 12 64 bit

Meshcentral version:
1.1.18

Node version:
v18.19.0

Remote device os:
Windows 11

**Comment by si458** [comment]
try updating to 1.1.20 ?

**Comment by adnan29979** [comment]
Updated to 1.1.20
Still the same.

**Comment by si458** [comment]
have you tried physically restarting the other computer?
also try uncommenting `agentPing` -> `_agentPing` as 18 seconds is VERY SHORT?

**Comment by adnan29979** [comment]
OMG, a restart of the Windows 11 agent worked. Thanks a lot. 

I set the agent ping to 45 seconds at first in my heavyweight server (350 agents). Because after a point in time, a lot of the devices were auto-disconnecting while I was working. The 45-second agent ping solved it. Then later down the line, some agents started showing this same issue before 30 seconds. Then I lowered it to 18 seconds and now that issue is gone. 

Anyways. Thanks a lot for the solution and for a very quick solution as well. I really love Meshcentral. A big fan of it. ❤❤
